storage/src/journal/authenticated.rs

```diff
@@ -400,23 +397,121 @@ where
             hasher,
         })
     }
+}
 
-    /// Durably persist the journal. This is faster than `sync()` but does not persist the MMR,
-    /// meaning recovery will be required on startup if we crash before `sync()` or `close()`.
-    pub async fn commit(&mut self) -> Result<(), Error> {
-        self.journal.commit().await.map_err(Error::Journal)
+impl<E, C, O, H> Contiguous for Journal<E, C, O, H>
+where
+    E: Storage + Clock + Metrics,
+    C: Contiguous<Item = O>,
+    O: Encode,
+    H: Hasher,
+{
+    type Item = O;
+
+    async fn append(&mut self, item: Self::Item) -> Result<u64, JournalError> {
+        let res = self.append(item).await.map_err(|error| match error {
+            Error::Journal(inner) => inner,
+            Error::Mmr(inner) => {
+                JournalError::Corruption(format!("MMR error during append: {}", inner))
```

Comment:
Perhaps not in this PR, but I wonder whether we should change the return type of `Contiguous` to allow for more natural-feeling values here. We use "corruption" elsewhere to mean that persisted data is borked, which isn't exactly what's happening here, but the "right" `JournalError` to return here isn't super apparent.